test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -111,22 +111,61 @@ protected override void Initialize()
             ML.AddStandardComponents();
         }
 
-        private string GetConfigurationDir()
-        {
+        private IEnumerable<string> GetConfigurationDirs()
+        {
+            // Sometimes different configuration will have combination.
+            // for example: one test can run on windows, have different result both
+            // on x64 vs x86 and dotnet core 3.1 vs others, so we have 4 combination:
+            // x64-netcore3.1, x86-netcore3.1, x64-rest, x86-rest. In some cases x64 vs x86
+            // have different results, in some cases netcore 3.1 vs rest have different results,
+            // the most complicate situation is 12 combinations (x64 vs x86, netcoreapp3.1 vs rest,
+            // win vs linux vs osx) have different results.
+            // So use list of string to return different configurations and test will try to search
+            // through this list and use the one file first found, make sure we don't have baseline file
+            // at different configuration folders.
+            var configurationDirs = new List<string>();
+
+            // Use netcore 3.1 result file if necessary.
+            // The small difference comes from CPUMath using different instruction set:
+            // 1. net framework and net core 2.1 uses CpuMathUtils.netstandard that uses SSE instruction set;
+            // 2. net core 3.1 uses CpuMathUtils.netcoreapp that uses AVX, SSE or direct floating point calculation
+            // depending on hardward avaibility.
+            // AVX and SSE generates slightly different result due to nature of floating point math.
+            // So Ideally we should adding AVX support at CPUMath native library, 
+            // use below issue to track: https://github.com/dotnet/machinelearning/issues/5044
+            // don't need netcoreapp21 as this is the default case
+            if (AppDomain.CurrentDomain.GetData("FX_PRODUCT_VERSION") != null)
+                configurationDirs.Add("netcoreapp31");
+
+            // don't need x64 as this is the default case
+            if (!Environment.Is64BitProcess)
+                configurationDirs.Add("x86");
+
             if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
-                return "osx-x64";
+                configurationDirs.Add("osx-x64");
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-                return "linux-x64";
+                configurationDirs.Add("linux-x64");
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 if (Environment.Is64BitProcess)
-                    return "win-x64";
+                {
+                    configurationDirs.Add("win-x64");
+
+                    if (AppDomain.CurrentDomain.GetData("FX_PRODUCT_VERSION") != null)
+                        configurationDirs.Add($"win-x64-netcoreapp31");
```

Comment:
Since we will be applying this for all the OSes, this can be factored out and checked once outside this if block

Reply:
yes, in theory we should apply this to all OSes but since we haven't investigate osx and linux case yet I don't want to make all the combination now because that seems little bit overkill and make all the combination will make the file exists check every time we try to reference baseline file that will likely impact our performance. 
I'm considering add osx and linux if this is necessary after investigation, make sense to you?

---
In reply to: [413231007](https://github.com/dotnet/machinelearning/pull/5047#discussion_r413231007) [](ancestors = 413231007)

Reply:
Yes, there might be a small delay because we are checking for an extra directory. But this delay happens in CI verification tests and doesn't represent the true performance of the algorithms. Those are covered by the benchmark tests.
I think it is okay to add it right now. But if you wish to add it later after investigation, I am okay with that too. 

---
In reply to: [413245167](https://github.com/dotnet/machinelearning/pull/5047#discussion_r413245167) [](ancestors = 413245167,413231007)

Reply:
Yeah, I think we better add any specific combination in future if necessary to keep the logic simple

---
In reply to: [413247756](https://github.com/dotnet/machinelearning/pull/5047#discussion_r413247756) [](ancestors = 413247756,413245167,413231007)